File: components/credentials/src/main/java/org/cloudfoundry/credhub/constants/CredentialType.java
package org.cloudfoundry.credhub.constants;

public enum CredentialType {
  PASSWORD("password"),
  CERTIFICATE("certificate"),
  VALUE("value"),
  RSA("rsa"),
  SSH("ssh"),
  JSON("json"),
  USER("user");

  public final String type;

  CredentialType(final String type) {
    this.type = type.toUpperCase();
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/constants/CredentialType.kt
package org.cloudfoundry.credhub.constants

enum class CredentialType private constructor(type: String) {
    PASSWORD("password"),
    CERTIFICATE("certificate"),
    VALUE("value"),
    RSA("rsa"),
    SSH("ssh"),
    JSON("json"),
    USER("user");

    val type: String

    init {
        this.type = type.uppercase()
    }

    companion object {
        fun generatableCredentialTypes(): List<CredentialType> {
            return listOf(
                PASSWORD,
                CERTIFICATE,
                USER,
                RSA,
                SSH
            )
        }
    }
}
